packages/server/src/stages/rewrite-imports/rewrite-imports.test.ts

```diff
@@ -0,0 +1,213 @@
+              export function someOtherFunction() {
+                return someFunction();
+              }
+            `,
+          },
+        ],
+        expectedOutput: [
+          {
+            path: normalize("/projects/blitz/blitz/app/pages/index.tsx"),
+            contents: `
+              export function someFunction() { return "foo"; }
+              export default function Index() { return <p>Hello World</p> }
+            `,
+          },
+          {
+            path: normalize("/projects/blitz/blitz/app/anyFile.ts"),
+            contents: `
+              import { someFunction } from "pages/index";
+              export function someOtherFunction() {
+                return someFunction();
+              }
+            `,
+          },
+        ],
+      }),
+    )
+  })
+
+  describe("an import from app/users/api/getUser", () => {
+    it(
+      "is rewritten to pages/api/getUser",
+      makeTest({
+        input: [
+          {
+            path: normalize("/projects/blitz/blitz/app/users/api/getUser.ts"),
+            contents: `
+              export function someFunction() { return "foo"; }
+              export default function Index() { return <p>Hello World</p> }
+            `,
+          },
+          {
+            path: normalize("/projects/blitz/blitz/app/anyFile.ts"),
+            contents: `
+              import { someFunction } from "app/users/api/getUser";
+              export function someOtherFunction() {
+                return someFunction();
+              }
+            `,
+          },
+        ],
+        expectedOutput: [
+          {
+            path: normalize("/projects/blitz/blitz/app/users/api/getUser.ts"),
+            contents: `
+              export function someFunction() { return "foo"; }
+              export default function Index() { return <p>Hello World</p> }
+            `,
+          },
+          {
+            path: normalize("/projects/blitz/blitz/app/anyFile.ts"),
+            contents: `
+              import { someFunction } from "pages/api/getUser";
+              export function someOtherFunction() {
+                return someFunction();
+              }
+            `,
+          },
+        ],
+      }),
+    )
+  })
+
+  describe("an import from app/users/queries/getUser", () => {
+    it(
+      "is rewritten to app/_resolvers/users/queries/getUser and app/users/queries/getUser",
+      makeTest({
+        input: [
+          {
+            path: normalize("/projects/blitz/blitz/app/anyFile.ts"),
+            contents: `
+import query, { someFunction } from "app/users/queries/getUser";
+export function someOtherFunction() {
+  return someFunction();
+}
+            `,
+          },
+        ],
+        expectedOutput: [
+          {
+            path: normalize("/projects/blitz/blitz/app/anyFile.ts"),
+            contents: `
+import query from "app/users/queries/getUser"
+import { someFunction } from "app/_resolvers/users/queries/getUser";
```

Comment:
Should we be testing for dynamic imports too? `const bar = import('foo/queries/bar')`

Reply:
How do these work? Does `import("foo/queries/bar")` import the default export (the query)? Or is it like an `import *`?

Because if they work like `import *`, I'm not so sure as to how we can rewrite those in a sensible way...

Reply:
> How do these work? Does import("foo/queries/bar") import the default export (the query)? Or is it like an import *?

I have no idea. Would have to look it up.

Reply:
Tried it, it imports the whole module.
![Screenshot 2020-11-16 at 11 10 08](https://user-images.githubusercontent.com/14912729/99240096-4ce12f00-27fc-11eb-8789-d99b779ebb17.png)

If you have an idea about how to rewrite these, I'm all ears. Otherwise, I'd say we let it slide for the moment (I don't think that a lot of people will import these dynamically, anyway)

Reply:
You can match the regex for it like this. 

```ts
/import\s?\(["']([^"']+)["']\);?/g
```

Reply:
Might have some corner cases.

Reply:
The problem is not detecting the import, but what to rewrite it to. See https://github.com/blitz-js/blitz/pull/1456#discussion_r523812784 for reference.

For imports from `api` and `pages` though, that's not a problem and I'll implement it.

Reply:
Hmm sounds like we're almost getting into AST territory...

Reply:
Still it can be done with regex.

Reply:
Actually then there is the issue when you take a var and use it later 

```ts
const foo = import('thing/blat/boo');
//
foo.default()
```

Reply:
Perhaps we should do a double whammy and use an ast to solve this as well as https://github.com/blitz-js/blitz/issues/734 ?

Reply:
Could you help me understand what improvements an AST would bring in this case? Currently, I don't see a way to rewrite * imports and dynamic imports so that they work _at all_. What would you replace the code example above with?

I'll take a shot at the conditional exports mentioned in https://github.com/blitz-js/blitz/pull/1456#discussion_r524200987, I think that could be successfull.

Reply:
Heureka ☺️ I guess I solved it with 15bb9cc. The `rpc` stage now generates both a `myQuery.ts` and a `myQuery.named.ts` that includes named exports.

In the rewrite-imports stage, imports that include any named imports are rewritten to use `myQuery.named.ts`.